src/Compiler/Checking/NameResolution.fs

```diff
@@ -530,14 +530,18 @@ let IsMethInfoPlainCSharpStyleExtensionMember g m isEnclExtTy (minfo: MethInfo)
 /// Get the info for all the .NET-style extension members listed as static members in the type.
 let private GetCSharpStyleIndexedExtensionMembersForTyconRef (amap: Import.ImportMap) m  (tcrefOfStaticClass: TyconRef) =
```

Comment:
I think it is consequential by the load to 
GetImmediateIntrinsicMethInfosOfType with the AccessorDomain  of searching all public types.

I will try to get output for which particular ttype it happens.

Reply:
Tanks @T-Gro . We appreciate the help 🙂

Reply:
The whole method **GetCSharpStyleIndexedExtensionMembersForTyconRef** called is being called on the type **TaskBuilderBase** , which happens to have a **Using** method which works with IAsyncDisposable.

This exists from netstandard21 onwards.

Reply:
Yes, much appreciated! Because I'm still pretty convinced this is part of a larger issue. (As I said earlier, I see this error on a weekly basis). It would be wonderful if a root cause could be found.

Reply:
Oh yes, it was.

This new code just meant hitting a code path (about assembly loading) that was not previously hit.

But the real issue is that the test suite was using an implicit (automagic) reference to Fsharp.Core library.
This however meant a mismatch between netstandard and FsharpCore version in specific tests, demonstrated by the inability to load IAsyncDisposable which is missing in netstandard20.

That is the theory at least, I now have the tests working locally and passed to CI checks.

This testsuite is not the best to troubleshoot, because it calls dotnet.exe ( = a lot of stuff happening without being to step trough it) and did not pass all references explicitly.


I first thought that the contents of GetCSharpStyleIndexedExtensionMembersForTyconRef  should be more defensive or swallow errors happening inside (returning None in the worst case), but that was a wrong direction.

Reply:
Nice 👌. Not longer need to rebuild the solution to fix the IAsyncDisposable thing . Thanks

Reply:
Progress has been made. the previous test is no longer failing. But now we have new ones :(. It seems this uncovering a wider issue. 

Would it make sense to move these tests to the ComponentTests to avoid this (automagic) reference to FSharp Core?

Reply:
(sorry for a few days of silence, I have been ill with my family)

Yes, I will tackle those tests and make them less flaky.